sdk/mobile-center-distribute/src/main/java/com/microsoft/azure/mobile/distribute/Distribute.java

```diff
@@ -643,6 +667,16 @@ else if (mUnknownSourcesDialog != null) {
                 }
             }
 
+            // If the in-app updates setup failed, and user ignores the failure, store the error
+            // message and also store the package hash that the failure occurred on. The setup
+            // will only be re-attempted the next time the app gets updated (and package hash changes).
+            String updateSetupFailedMessage = PreferencesStorage.getString(PREFERENCE_KEY_UPDATE_SETUP_FAILED_MESSAGE_KEY);
+            if (updateSetupFailedMessage != null){
+                MobileCenterLog.debug(LOG_TAG, "In-app updates setup failure detected.");
+                showUpdateSetupFailedDialog(updateSetupFailedMessage);
```

Comment:
should this block be moved into `DeepLinkActivity.java` into the area where `storeUpdateSetupFailedParameter` is being called, and then we no longer need to store the update setup failure message?

Reply:
@JessicaYeh 
Initially I supposed to do so but the problem with calling this from `DeepLinkActivity` is that `mForegroundActivity` is `null` at that moment, so we can't build dialog at that time. 
That is also the reason I decided to store `errorMessage` with `PREFERENCE_KEY_UPDATE_SETUP_FAILED_MESSAGE_KEY` in order to be able to retrieve it later to show error dialog with this message. 
Im open to any suggestions if you see how we can handle this otherwise.

@jaelim-ms  Could you also please take a look at this and advise?